src/post/post.repository.knex.ts

```diff
@@ -23,6 +23,118 @@ export class KnexPostRepository {
         private readonly events: AsyncEvents,
     ) {}
 
+    async getById(id: Post['id']): Promise<Post | null> {
```

Comment:
This function looks pretty similar to `getByApId`, and there’s quite a bit of duplicated code. Might be a good idea to refactor some of the shared parts into common utility methods. We an also reuse them in other get queries later on.

Reply:
That's a good point, thanks for pointing it out - I'll clean it up before merging!